src/worker/lambda/handlerPuller.go

```diff
@@ -33,6 +33,62 @@ type CacheEntry struct {
 	path    string // where code is extracted to a dir
 }
 
+func ccopy(src, dest string) error {
+	info, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	if info.IsDir() {
+		return filepath.Walk(src, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			relPath, err := filepath.Rel(src, path)
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				return os.MkdirAll(filepath.Join(dest, relPath), info.Mode())
+			}
+
+			return copyFile(path, filepath.Join(dest, relPath))
+		})
+	}
+
+	return copyFile(src, dest)
+}
+
+func copyFile(src, dest string) error {
+	srcFile, err := os.Open(src)
```

Comment:
How does this work if "src" is a symlink?  Do we copy the link itself, or do we copy the data?  Copying the data seems dangerous -- somebody could add a symlink pointing to whatever on the host machine and get it inside their lambda.  Please check and verify (and leave a comment explaining the security of it).

Some tests with symlinks are a good idea too.

Reply:
That is indeed a situation not considered at all, if the src file is a symlink, it will trace to the content and copy it and make this function prone to attack. Added a check in the new version of code to check if the src is a symlink and right now copy the symlink instead of the content (or maybe we should ignore symlink?).

Reply:
Thanks for the security fix!  Can we do it faster, though?  I see some folks combine the check with an OpenFile call, like this: https://blog.trailofbits.com/2020/11/24/smart-and-simple-ways-to-prevent-symlink-attacks-in-go.

Reply:
Thanks! I just change the symbolic link check to O_NOFOLLOW when opening the file